### DISCUSSION TITLE
Ticker-like behavior and `RescheduleMode`

### DISCUSSION THREAD

**Discussion by husam-e** [go-co-op/gocron#571] [Q&A] [answered]
Hello,

I am trying to achieve behavior similar to a [Ticker](https://pkg.go.dev/time#NewTicker):

> The ticker will adjust the time interval or drop ticks to make up for slow receivers.

Is there a way to achieve this? The docs suggest that using `SetMaxConcurrentJobs(1, gocron.RescheduleMode)` should do this, but based on local testing, it seems to do the same thing as `SingletonMode` and `SetMaxConcurrentJobs(1, gocron.WaitMode)`, where if a Job takes longer than the scheduler interval for a few iterations, once it gets back under the scheduler interval's duration, a bunch of jobs execute immediately until all the missed jobs are run, instead of just waiting till the next "tick" to schedule the next job.

Asked this in the Slack channel as well: [thread](https://gophers.slack.com/archives/CQ7T0T1FW/p1695133635284479)

**Comment by JohnRoesler** [comment, marked as answer]
#572 Thanks for pointing this out. The functionality exists with limit max concurrent runs and reschedule mode. Fixed in the latest release

**Reply by husam-e** [reply]
Thanks! Working as expected now.
